packages/dds/sequence/src/test/intervalCollection.spec.ts

```diff
@@ -1731,5 +1759,166 @@ describe("SharedString interval collections", () => {
 				assert.equal(collection.detachIndex(mockIntervalIndex), false);
 			});
 		});
+
+		describe("support querying intervals with endpoint in a range", () => {
+			let helpers: IIntervalHelpers<Interval>;
+			let endInRangeIndex: EndInRangeIndex<Interval>;
+			let compareFn;
+			let results;
+
+			beforeEach(() => {
+				helpers = {
+					compareEnds: (a: Interval, b: Interval) => a.end - b.end,
+					create: createInterval,
+				};
+				endInRangeIndex = new EndInRangeIndex(helpers);
+				// sort the query result by the interval endpoint value
+				compareFn = (a: Interval, b: Interval) => {
+					if (a.end === b.end) {
+						return a.start - b.start;
+					}
+					return a.end - b.end;
+				};
+			});
+
+			it("can not return result if the input does not meet specified requirement", () => {
+				results = endInRangeIndex.findIntervalsWithEndInRange(1, 1);
+				assert.equal(
+					results.length,
+					0,
+					"Should not return anything once the index is empty",
+				);
+
+				endInRangeIndex.add(new Interval(1, 1));
+				endInRangeIndex.add(new Interval(1, 3));
+
+				results = endInRangeIndex.findIntervalsWithEndInRange(2, 1);
+				assert.equal(
+					results.length,
+					0,
+					"The start value should not be lower than the end value",
```

Comment:
message doesn't make sense to me. also, why query the range `[2, 1]`? seems odd

Reply:
This is to handle corner cases and proactively prevent any unexpected behavior when performing searches in the tree. And we could include the case where the endpoints values are negative as well. 

![image](https://github.com/microsoft/FluidFramework/assets/86381878/079753c7-5d9b-4f60-a09a-98147eaaf426)